src/tactic/interactive_expr.lean

```diff
@@ -390,8 +393,16 @@ tc.stateless $ λ ft, do
   let lcs := group_local_collection lcs,
   lchs ← lcs.mmap (λ lc, do
     lh ← local_c lc,
-    let ns : list (html γ) := lc.locals.map $ λ n,
-      h "span" [cn "goal-hyp b pr2", key n.local_uniq_name] [html.of_name n.local_pp_name],
+    ns ← lc.locals.mmap (λ n, do -- extract into function?
+      cls ← is_class lc.type,
+      let var_color : attr γ := if ¬ cls
+        then attr.style [("color", "#cc7a00")]  -- "goal-hyp"
+        else if frozen
+          then if n ∈ is'
+            then attr.style [("color", "#2aa198")]  -- "goal-hyp-inst"
+            else attr.style [("font-style", "italic")] -- "goal-hyp-noninst"
+        else attr.style [("color", "#dc322f")], -- "goal-hyp-unfrozen"
```

Comment:
Can avoid the `map` stuff above with
```suggestion
        else match is with
        | some is' := if n ∈ is'
            then attr.style [("color", "#2aa198")]  -- "goal-hyp-inst"
            else attr.style [("font-style", "italic")] -- "goal-hyp-noninst"
        | none := attr.style [("color", "#dc322f")], -- "goal-hyp-unfrozen"
        end
```

Reply:
The goal was performance, because currently we only calculate `frozen` once, rather that at each iteration of the loop.

Reply:
Your version is `option.cases_on`+`n`*`bool.cases_on`, my version is `n`*`option.cases_on`. I don't expect any measurable performance difference.